File: ios/Samples/Showcase-ObjectiveC/LiferayScreens-Showcase-ObjectiveC/AssetDisplayViewController.m
//
//  AssetDisplayViewController.m
//  LiferayScreens-Showcase-ObjectiveC
//
//  Created by Victor Galán on 03/02/2017.
//  Copyright © 2017 liferay. All rights reserved.
//

@import LiferayScreens;
#import "UserView.h"
#import "LiferayLogger.h"
#import "AssetDisplayViewController.h"

@interface AssetDisplayViewController () <AssetDisplayScreenletDelegate>

@property (weak, nonatomic) IBOutlet AssetDisplayScreenlet *screenlet;

@end

@implementation AssetDisplayViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.screenlet.delegate = self;
    self.screenlet.presentingViewController = self;
    self.screenlet.assetEntryId = [LiferayServerContext longPropertyForKey:@"assetDisplayEntryId"];

    [self.screenlet load];
}

- (void)screenlet:(AssetDisplayScreenlet *)screenlet onConfigureScreenlet:(BaseScreenlet *)childScreenlet onAsset:(Asset *)asset {
    LiferayLog(childScreenlet, asset);
}

- (void)screenlet:(AssetDisplayScreenlet *)screenlet onAssetResponse:(Asset *)asset {
	LiferayLog(asset);
}

- (void)screenlet:(AssetDisplayScreenlet *)screenlet onAssetError:(NSError *)error {
	LiferayLog(error);
}

- (UIView *)screenlet:(AssetDisplayScreenlet *)screenlet onAsset:(Asset *)asset {
    NSArray *keys = [asset.attributes[@"object"] allKeys];

    if ([keys containsObject:@"user"]) {
        UserView *userView = [[UserView alloc] init];
        NSDictionary *object = asset.attributes[@"object"];

        userView.user = [[User alloc] initWithAttributes:object[@"user"]];

        return userView;
    }

    return nil;
}

@end

File: ios/Samples/Showcase-swift/LiferayScreens-Showcase-Swift/ViewControllers/AssetDisplayViewController.swift
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */
import UIKit
import LiferayScreens


class AssetDisplayViewController: UIViewController, AssetDisplayScreenletDelegate {
	
	@IBOutlet var screenlet: AssetDisplayScreenlet?
	
	var entryId: Int64?
	
	override func viewDidLoad() {
		super.viewDidLoad()

		if let id = entryId {
			self.screenlet?.assetEntryId = id
		}
		self.screenlet?.delegate = self
		self.screenlet?.presentingViewController = self
		self.screenlet?.load()
	}
	
	func screenlet(screenlet: AssetDisplayScreenlet,
	               onAssetEntryResponse assetEntry: Asset) {
		print("DELEGATE: onAssetEntryResponse -> \(assetEntry)\n");
	}
	
	func screenlet(screenlet: AssetDisplayScreenlet,
	               onAssetEntryError error: NSError) {
		print("DELEGATE: onAssetEntryError -> \(error)\n");
	}

	func screenlet(screenlet: AssetDisplayScreenlet, onAsset asset: Asset) -> UIView? {
		if let type = asset.attributes["object"]?.allKeys.first as? String {
			if type == "user" {
				let vc = self.storyboard?.instantiateViewControllerWithIdentifier("UserDisplay") as? UserDisplayViewController
				if let userVc = vc {
					self.addChildViewController(userVc)
					screenlet.addSubview(userVc.view)
					userVc.view.frame = screenlet.bounds
					userVc.user = User(attributes: asset.attributes)
				}
			}
		}
		return nil
	}
}
